### ISSUE TITLE
[Firebase] permission_denied error on Firebase sign out

### ISSUE THREAD

**Issue by t2ym** [t2ym/live-localizer#69]
[Firebase] permission_denied error on Firebase sign out

It seems that the Firebase client tries to flush data to the cloud storage just after it has signed out.  Does the websocket connection for Firebase know the sign out process properly?  ~~Why is the issue reproducible only with Polymer 1.x?~~

| polymer | polymerfire | firebase | reproducible | browsers |
|:--------:|:-----------:|:--------:|:-------:|:-----:|
| 1.9.1 | 2.1.0 | 4.1.2 | ✔︎ | Chrome 59, Firefox 54 |
| 2.0.1 | 2.1.0 | 4.1.2 | ✔︎ | Chrome 59, Firefox 54 |
| 1.9.1 | 0.9.7 | 3.9.0 | ✔︎ | Chrome 59, Firefox 54 |

```
sign out
polymer.html:1284 error Error: permission_denied at /users/w8ZZ1noWBtWEHB7ZheFNonKrJFb2/files/ja: Client doesn't have permission to access the desired data.
    at G (Query.js:270)
    at Object.G (Query.js:358)
    at Lg (Repo.js:96)
    at Ag.g.wd (SyncTree.js:366)
    at og.wd (DataSnapshot.js:217)
    at Yf.Xf (DataSnapshot.js:142)
    at ag (PersistentConnection.js:465)
    at WebSocket.Ia.onmessage (PersistentConnection.js:385)
polymer.html:1284 error Error: permission_denied at /users/w8ZZ1noWBtWEHB7ZheFNonKrJFb2/settings: Client doesn't have permission to access the desired data.
    at G (Query.js:270)
    at Object.G (Query.js:358)
    at Lg (Repo.js:96)
    at Ag.g.wd (SyncTree.js:366)
    at og.wd (DataSnapshot.js:217)
    at Yf.Xf (DataSnapshot.js:142)
    at ag (PersistentConnection.js:465)
    at WebSocket.Ia.onmessage (PersistentConnection.js:385)
```

**Comment by t2ym** [comment]
It seems no data are lost on sign out as the data have been sync'ed with the server before the client signs out.  Needs more verification, though.

**Comment by t2ym** [comment]
### Root Cause:
Firebase client tries to flush the client cache even after the client signs out.

### Fixes:
- (In `firebase-storage`) `destroy()` `firebase-document` elements (`$.settings`, `$.document`) if their `path`s are set
- (In test) Wait 2 seconds before sign out via `firebase-storage.onTap()` so that the client caches are flushed

**Comment by t2ym** [comment]
### Warning:
Immediate sign out after writing data to the server may cause the error, which makes the tests fail but should not affect manual operations.

**Comment by t2ym** [comment]
### Limitations:
On Firefox, the fixes are ineffective for `polymerfire 0.9.7` with `firebase 3.9.0` in `1.x-old` dependencies
